src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/ProjectState.cs

```diff
@@ -196,26 +196,44 @@ RazorProjectEngine CreateProjectEngine()
 
     public VersionStamp ConfigurationVersion { get; }
 
-    public ProjectState WithAddedHostDocument(HostDocument hostDocument, TextLoader loader)
+    public ProjectState AddDocument(HostDocument hostDocument, SourceText text)
```

Comment:
I would have thought the "With" part of the name made sense, since this is not mutating. Curious your reasoning?

(or its entirely possible this makes sense in context of a future commit)

Reply:
Not just to match Roslyn, but also because it's the correct and consistent naming scheme used throughout .NET. For example, modifying an immutable collection is done via "Add", "Remove", "Set", etc. methods while setting properties, such as "Comparer", is handled by "With" methods. Also, the whole "WithAddedDocument" sort of naming is a crisis of mixing English tenses in odd ways. :smile: